### DISCUSSION TITLE
How to customize two different sections at same time ?

### DISCUSSION THREAD

**Discussion by Marlony-Gabriel** [squidfunk/mkdocs-material#7395] [Q&A]
Hi.
I need to use two different scripts. Onde needs to be in the **head** section and the other needs to be on all pages.
I tried the following solution, but It's not working

```
  mkdocs.yml configuration
      theme:
           - name: material
           - custom_dir: overrides
```

overrides folder
    partials
        | - source-file.html (used for mkdocs-git-revision-date-localized-plugin)
    main.html

main.html code 

```
{% extends "base.html" %}

{% block extrahead %}
<script src="//code.jivosite.com/widget/1111111111" async></script>
{% endblock %}

{% block header %}
  <script>
     "my code here"
  </script
  {{ super() }}
  <!-- Add scripts that need to run afterwards here -->
{% endblock %}
```
My code is suposed to create a dynamic content search, for pages with a lot of contents
like this: https://centraldeajuda.serpro.gov.br/acessohod/faqs/outrasduvidas/
The main ideia is make the navigation easier, allowing the users to find the topic of your interest without need to read the entire page.
[mycode.txt](https://github.com/user-attachments/files/16362459/mycode.txt)

**Comment by kamilkrzyskow** [comment]
Hello @Marlony-Gabriel,
I'm not sure I understood the idea:
> I need to use two different scripts. One needs to be in the head section and the other needs to be on all pages.

`main.html` is the entry point for all rendered Markdown pages, unless you change the template.
Therefore, both the `extrahead` and `header` blocks contents will be added to every page without additional conditionals.

Given your code from `main.html` I see no need to override the `source-file.html` partial, and I don't see any syntax errors.
> I tried the following solution, but It's not working

So you'd have to describe in more detail how it doesn't work. Are you certain that the scripts don't show up in the HTML source code?
I recommend starting with a clean slate `mkdocs new .` and step-by-step add the overrides again. You can also share the [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) zip archive.

I can't help you with your custom JavaScript code, if the issue is inside then you'll have to debug it yourself.
